cs/testing-with-travis.texy

```diff
@@ -0,0 +1,234 @@
+
+Abyste předešli podobným zážitkům, ukážeme si, jak průběžně testovat pomocí populárního nástroje [Travis CI](https://travis-ci.org/):
+
+- nastavení testovacího prostředí
+- instalace závislostí pomocí Composeru
+- spouštění testů pomocí Nette/Tester
+- nastavení dalších služeb
+- [TL;DR](#Výsledný soubor)
+- integrace s GitHubem
+\--
+
+Jak funguje Travis CI
+=====================
+
+Travis CI (dále jen Travis) je velmi populární nástroj pro automatizaci [průběžné integrace](http://cs.wikipedia.org/wiki/Průběžná_integrace). Je plně integrován do prostředí GitHubu.
+
+Travis se nastavuje pomocí konfiguračního souboru `.travis.yml` umístěného v kořenovém adresáři projektu. Obsahuje několik bloků, které si podrobněji popíšeme.
+
+Bloky jsou spouštěny v pořadí `before_install`, `install`, `before_script`, `script`, `after_success` nebo `after_failure` a nakonec `after_script`. Povinný je pouze blok `script`.
+
+K výsledku testu můžete v nastavení přistoupit pomocí proměnné `$TRAVIS_TEST_RESULT`.
+
+Pro podrobnější informace si přečtěte [oficiální dokumentaci |http://docs.travis-ci.com].
+
+Vytvoření .travis.yml
+=====================
+
+Nastavení jazyka
+----------------
+
+Nejprve je potřeba pomocí `language` Travis informovat, pro jaký jazyk má prostředí nastavit. Na dalších řádcích uvádíme, pro jaké verze PHP budeme chtít náš projekt otestovat. Testy pak budou puštěny pro každou verzi zvlášť. Protože Nette Tester podporuje i [HHVM](http://hhvm.com), můžeme ho přidat také.
+
+/--
+language: php
+
+php:
+  - 5.3
+  - 5.4
+  - 5.5
+  - 5.6
+  - hhvm
+\--
+ 
+Proměnné prostředí
+------------------
+
+V sekci `env` nastavujeme proměnné prostředí. Pro každou odrážku se skripty provedou samostatně a v jedné odrážce lze definovat více proměnných. My budeme používat proměnnou `TESTER_PHP_BIN`, abychom pro HHVM mohli testy pouštět s parametrem `-p hhvm`. Hodnotu proměnné získáme pomocí prefixu `$`.
+
+.[caution]
+Testy pro HHVM by fungovali i s možností `-p php`, Travis vytváří symlink pro `hhvm`. Nelze ovšem vynechat úplně, protože Travis ve výchozím nastavení používá binárku `php-cgi`, která už pro HHVM dostupná není.
+
+/--
+env:
+  - TESTER_PHP_BIN="php"
+  - TESTER_PHP_BIN="hhvm"
+\--
+
+Instalace závislostí
+--------------------
+
+.[caution]
+Návod předpokládá, že závislosti vašeho projektu jsou spravovány pomocí nástroje Composer a v kořenovém adresáři obsahuje `composer.json`. Zároveň musíte mít v závislostech uvedený nette/tester. Jak nastavit Composer najdete v kapitole [#Composer].
+
+V bloku `install` nainstalujeme pomocí Composeru závislosti našeho projektu. Každá odrážka znamená samostatný příkaz. Composer defaultně nainstaluje i dev závislosti. Dobrým zvykem je použít `--no-interaction`, protože Travis si s Composerem neumí povídat. Použijeme i `--prefer-source` kvůli [chybě Composeru |https://github.com/composer/composer/issues/1314].
```

Comment:
toto je imo zavadeji, to neni chyba composeru, to uz spis githubu. proste bych to vysvetlil proc.